consultation_analyser/consultations/views/answers.py

```diff
@@ -18,150 +19,285 @@ class DataDict(TypedDict):
-                )
-            )
-            .prefetch_related(
-                Prefetch("evidencerichmapping_set", to_attr="prefetched_evidencerichmappings")
-            )
-            .prefetch_related(
-                Prefetch("sentimentmapping_set", to_attr="prefetched_sentimentmappings")
-            )
-        )
 
-        respondents = (
-            models.RespondentOld.objects.filter(
-                id__in=Subquery(answers.values_list("respondent_id", flat=True))
-            )
-            .prefetch_related(
-                Prefetch(
-                    "answer_set", queryset=answers, to_attr="prefetched_answers"
-                )  # Prefetch the necessary answers
-            )
-            .order_by("pk")
-        )
+def get_cached_theme_summary(question_id: str, filters_hash: str) -> list[dict] | None:
+    """Get cached theme summary or None if not cached"""
+    cache_key = f"theme_summary:{question_id}:{filters_hash}"
+    return cache.get(cache_key)
+
+
+def set_cached_theme_summary(question_id: str, filters_hash: str, data: list[dict], timeout: int = 300):
+    """Cache theme summary for 5 minutes"""
+    cache_key = f"theme_summary:{question_id}:{filters_hash}"
+    cache.set(cache_key, data, timeout)
+
+
+def get_filters_hash(filters: FilterParams) -> str:
+    """Generate a hash for caching based on filter parameters"""
+    return hashlib.md5(str(sorted(filters.items())).encode()).hexdigest()
+
+
+def build_respondent_data(respondent: models.Respondent, response: models.Response) -> dict:
+    """Extract respondent data building to separate function"""
+    data = {
+        "id": f"response-{respondent.identifier}",
+        "identifier": str(respondent.identifier),
+        "sentiment_position": "",
+        "free_text_answer_text": response.free_text or "",
+        "demographic_data": respondent.demographics or {},
+        "themes": [],
+        "multiple_choice_answer": [response.chosen_options] if response.chosen_options else [],
+        "evidenceRich": False,
+        "individual": respondent.demographics.get("individual", False),
+    }
+    
+    if hasattr(response, "annotation") and response.annotation:
+        annotation = response.annotation
+        
+        if annotation.sentiment:
+            data["sentiment_position"] = annotation.sentiment
+        
+        if annotation.evidence_rich == models.ResponseAnnotation.EvidenceRich.YES:
+            data["evidenceRich"] = True
+        
+        # Add themes (already prefetched)
+        data["themes"] = [
+            {
+                "id": theme.id,
+                "stance": None,  # Stance is no longer stored in new models
+                "name": theme.name,
+                "description": theme.description,
+            }
+            for theme in annotation.themes.all()
+        ]
+    
+    return data
+
+
+def get_demographic_options(consultation: models.Consultation) -> dict[str, list]:
```

Comment:
One to consider - maybe check when we have loads of data.

Could this be time consuming when we have loads of data?

Possible solution would be to store this dictionary on the consultation as e.g. demographic_data_options? (or a better name)